lib/search/l/lfind.h

```diff
@@ -12,19 +12,13 @@
 #include <stddef.h>
 
 #include "search/cmp/cmp.h"
-#include "typetraits.h"
 
-#include <assert.h>
 
-
-#define LFIND(k, a, n)                                                \
+#define LFIND(T, k, a, n)                                             \
 ({                                                                    \
-	__auto_type  k_ = k;                                          \
-	__auto_type  a_ = a;                                          \
-                                                                      \
-	static_assert(is_same_typeof(k_, a_), "");                    \
-                                                                      \
-	(typeof(k_)) lfind_(k_, a_, n, sizeof(*k_), CMP(typeof(k_))); \
+	_Generic(k, T *: 0, const T *: 0);                            \
```

Comment:
Do I understand correctly that the _Generic() here is to make sure, by not
having a default, that if k is neither T * nor const T *, then compilation
will fail?